sklearn/linear_model/base.py

```diff
@@ -273,6 +273,11 @@ def _set_intercept(self, X_offset, y_offset, X_scale):
         """Set the intercept_
         """
         if self.fit_intercept:
+            if isinstance(self.coef_, np.ndarray):
```

Comment:
Could you clarify why `X_offset` etc are of dtype np.float64?

In any case, since `self.coef_` is always a numpy array, you can get the dtype without doing the if check, right?